cmd/sfu/main.go

```diff
@@ -100,11 +100,16 @@ func main() {
 		return
 	}
 
-	// Create a router to route incoming To-Device messages to the right conference.
-	routerChannel := routing.NewRouter(matrixClient, connectionFactory, config.Conference)
+	// Create a channel which we'll use to send events to the router.
+	matrixEvents := make(chan *event.Event)
+
+	// Start a router that will receive events from the matrix client and route them to the appropriate conference.
+	routing.RunRouter(matrixClient, connectionFactory, matrixEvents, config.Conference)
 
 	// Start matrix client sync. This function will block until the sync fails.
 	matrixClient.RunSyncing(func(e *event.Event) {
-		routerChannel <- e
+		matrixEvents <- e
 	})
```

Comment:
`matrixClient.RunSyncing` running endless until a panic happen. Is that right?

We could avoid this if we change the signature like this:
```go
	matrixEvents, errc := s.matrixClient.RunSyncing()
	routing.RunRouter(matrixClient, connectionFactory, matrixEvents, config.Conference)
	if err := <-errc; err != nil {
		log(err)
	}
```

Reply:
Ah, I see. Do I get it right that the advantage is that we don't panic from inside of a package, but do it in `main` instead?

Reply:
We could decide that by what kind of error occurred. I think we should avoid panic if not needed.

Reply:
Yeah, makes sense, though we only have one type of error. But please confirm that my understanding is correct: the reason why we're doing this is that it would allow us to avoid panic inside a package? - If so, I agree, but couldn't we solve it easier by just returning this error instead of panicking? I.e. I can slightly change the `RunSyncing()` so that it does not panic once syncing stopped, but instead just returns the error back to the caller.

Reply:
`RunSyncing()` acts as the main loop. `errc` is a kind of done channel that could interrupt the loop gracefully. Additionally `errc` contains the reason for the termination. With `nil` it was an intentional abort and with `error` an abort because an issue.

Reply:
Right. But we could simply return an error instead of a panic which would be the same effect, right?

Anyway, I pushed a couple of new changes, so that:

1. We don't panic anymore from inside the signaling package, we return an error instead (to be consistent with the API of the Matrix SDK there; it seems like a separate `errc` is not required in that particular case, but could be added on top of it if we ever need it?).
2. I don't store the `done` inside a conference state anymore (as you pointed out, we only use it from the `processMessages()` loop to close it once the loop is over, so I just pass it as a local variable. That way no other function can [by accident] close a channel that it's not supposed to close 👍 

Ptal 🙂